iOS Client/Store/MBStore.swift

```diff
@@ -38,6 +38,19 @@ class MBStore: NSObject {
         return performFetch(managedContext: persistentContainer.viewContext, fetchRequest: fetchRequest) as? [MBArticle] ?? []
     }
     
+    func getBookmarkedArticles(persistentContainer: NSPersistentContainer) -> [MBArticle]{
+        let fetchRequest = NSFetchRequest<NSManagedObject>(entityName: MBArticle.entityName)
+        let predicate = NSPredicate(format: "bookmarked == TRUE")
+        fetchRequest.predicate = predicate
+        return performFetch(managedContext: persistentContainer.viewContext, fetchRequest: fetchRequest) as? [MBArticle] ?? []
+
+    }
+    
+    func bookmark(article: MBArticle, persistentContainer: NSPersistentContainer) throws {
```

Comment:
hmmm, this is assuming the MBArticle NSManagedObject is linked to the Persistent Container's View context. This happens to be the case, but it's a little dangerous to assume so.

I think you can just pass the article here, and use its managedObjectContext property to get the context, on which you can call save().